### PR TITLE
Add support for Python 3.11 and 3.12.

### PR DESCRIPTION
**Pull request by simonkagwi**
This PR updates the tox config and the GitHub Actions workflow to test against Python 3.11 and 3.12, and adds the two versions to the list of supported Python versions in setup.py and README.rst. It also fixes an error building docs due to an outdated (pre-Sphinx 1.0) format for the [`intersphinx_mapping`](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#confval-intersphinx_mapping) configuration, which is no longer accepted in Sphinx 8.0.0+.

Tox summary:
bread_tox

### CODE REVIEW

**Comment by simonkagwi** [comment]
Added [pyupgrade](https://github.com/asottile/pyupgrade) and [django-upgrade](https://github.com/adamchainz/django-upgrade) pre-commit hooks and dropped support for Python 3.8 which is now officially EOL.

Tox summary:
bread_tox2

**Comment by simonkagwi** [comment]
@tobiasmcnulty I've dropped support for Django 3.2 and 4.1 and added support for Django 5.0 and 5.1.

bread_tox3

File: bread/bread.py [hunk 1]
```diff
@@ -69,8 +69,8 @@ def get_full_perm_name(self, short_name):
 
     def __init__(self, *args, **kwargs):
         # Make sure the permission needed to use this view exists.
-        super(BreadViewMixin, self).__init__(*args, **kwargs)
-        perm_name = "%s_%s" % (
+        super().__init__(*args, **kwargs)
+        perm_name = "{}_{}".format(
```
Comment (Mariatta) [inline bread/bread.py:73]: I noticed this was not updated to use f-strings. Is this done by pyupgrade?

Comment (simonkagwi) [reply bread/bread.py:73]: Yes, that was done by pyupgrade

**Review by Mariatta — APPROVED** [review]
Thanks!

**Pull request merged**